skorch/cli.py

```diff
@@ -0,0 +1,335 @@
+            param, prefix=prefix, defaults=defaults))
+        yield " " * indent + first
+        for line in rest:
+            yield " " * 2 * indent + line
+
+
+def _get_help_for_estimator(prefix, estimator, defaults=None):
+    """Yield help lines for the given estimator and prefix."""
+    defaults = defaults or {}
+    estimator = _extract_estimator_cls(estimator)
+    yield "<{}> options:".format(estimator.__name__)
+
+    doc = ClassDoc(estimator)
+    yield from _get_help_for_params(
+        doc['Parameters'],
+        prefix=prefix,
+        defaults=defaults,
+    )
+    yield ''  # add a newline line between estimators
+
+
+def print_help(model, defaults=None):
+    """Print help for the command line arguments of the given model.
+
+    Parameters
+    ----------
+    model : sklearn.base.BaseEstimator
+      The basic model, e.g. a ``NeuralNet`` or sklearn ``Pipeline``.
+
+    defautls : dict or None (default=None)
+      Optionally, change the default values to use custom
+      defaults. Commandline arguments have precedence over defaults.
+
+    """
+    defaults = defaults or {}
+
+    print("This is the help for the model-specific parameters.")
+    print("To invoke help for the remaining options, run:")
+    print("python {} -- --help".format(sys.argv[0]))
+    print()
+
+    lines = (_get_help_for_estimator(prefix, estimator, defaults=defaults) for
+             prefix, estimator in _yield_estimators(model))
+    print('\n'.join(chain(*lines)))
+
+
+def parse_args(kwargs, defaults=None):
+    """Apply command line arguments or show help.
+
+    Use this in conjunction with the fire library to quickly build
+    command line interfaces for your scripts.
+
+    This function returns another function that must be called with
+    the estimator (e.g. ``NeuralNet``) to apply the parsed command
+    line arguments. If the --help option is found, show the
+    estimator-specific help instead.
+
+    Examples
+    --------
+    Content of my_script.py:
+
+    >>> def main(**kwargs):
+    >>>     X, y = get_data()
+    >>>     my_model = get_model()
+    >>>     parsed = parse_args(kwargs)
+    >>>     my_model = parsed(my_model)
+    >>>     my_model.fit(X, y)
+    >>>
+    >>> if __name__ == '__main__':
+    >>>     fire.Fire(main)
+
+    Parameters
+    ----------
+    kwargs : dict
+      The arguments as parsed by fire.
+
+    defautls : dict or None (default=None)
+      Optionally, change the default values to use custom
+      defaults. Commandline arguments have precedence over defaults.
+
+    Returns
+    -------
+    print_help_and_exit : callable
+      If --help is in the arguments, print help and exit.
+
+    set_params : callable
+      If --help is not in the options, apply command line arguments to
+      the estimator and return it.
+
+    """
+    defaults = defaults or {}
+
+    def print_help_and_exit(estimator):
+        print_help(estimator, defaults=defaults)
+        sys.exit()
```

Comment:
Maybe use `exit(1)` to indicate non-normal termination?

Reply:
But should the error code not be 0 for a normal help option?

Reply:
Well, not sure, the usual tools don't seem to agree on that issue:
```
$ cat -h 2>/dev/null; echo $?
1
$ sed -h 2>/dev/null; echo $?
1
$ dd -h 2>/dev/null; echo $?
1
$ grep --help >/dev/null; echo $?
0
$ make -h >/dev/null; echo $?
0
$ bzcat -h; echo $?
0
$ bash --help; echo $?
0
```

But it might be good to exit with an error code for cases when deploying the script to distributed/automated training. An illegal option might be detected sooner if the script fails to execute due to an unknown option or a left-over `-h`.